app/forms/provider_interface/interview_form.rb

```diff
@@ -0,0 +1,81 @@
+module ProviderInterface
+  class InterviewForm
+    include ActiveModel::Model
+
+    attr_accessor :day, :month, :year, :time, :location, :additional_details, :application_choice, :provider_id, :current_provider_user
+
+    validates :application_choice, :current_provider_user, presence: true
+    validate :date_is_valid?
+    validate :date_and_time_in_future, if: ->(form) { form.date_is_valid? && form.time_is_valid? }
+    validate :time_is_valid?
+    validates :location, presence: true
+
+    def date_and_time
```

Comment:
This doesn't quite work. This method is unable to parse times like `3pm` correctly, so we should use `Time.zone.parse`. This does mean we'll need to first make the date and time into a string like `2021-1-1 3:00pm`